model/jpa/src/main/java/org/keycloak/organization/jpa/JpaOrganizationProvider.java

```diff
@@ -76,21 +78,22 @@ public OrganizationModel create(String name, Set<String> domains) {
             throw new ModelDuplicateException("A organization with the same name already exists.");
         }
 
-        OrganizationEntity entity = new OrganizationEntity();
-        entity.setId(KeycloakModelUtils.generateId());
+        OrganizationAdapter adapter = new OrganizationAdapter(realm, this);
 
-        GroupModel group = createOrganizationGroup(entity.getId());
-
-        entity.setGroupId(group.getId());
-        entity.setRealmId(realm.getId());
-        entity.setName(name);
-        entity.setEnabled(true);
+        try {
+            session.setAttribute(OrganizationModel.class.getName(), adapter);
+            GroupModel group = createOrganizationGroup(adapter.getId());
 
-        em.persist(entity);
+            adapter.setGroupId(group.getId());
+            adapter.setName(name);
+            adapter.setEnabled(true);
 
-        OrganizationAdapter adapter = new OrganizationAdapter(realm, entity, this);
+            em.persist(adapter.getEntity());
 
-        adapter.setDomains(domains.stream().map(OrganizationDomainModel::new).collect(Collectors.toSet()));
+            adapter.setDomains(domains.stream().map(OrganizationDomainModel::new).collect(Collectors.toSet()));
+        } finally {
+            session.removeAttribute(OrganizationModel.class.getName());
```

Comment:
Do we need to remove the attribute from the session? Can we add it in OrganizationResource and in the constructor of the JpaOrganizationProvider and leave it there for the lifespan of the session?

Reply:
Yes, it does not make sense to keep it there after this method is called.